components/selection/selection-checkbox.js

```diff
@@ -0,0 +1,122 @@
+import '../inputs/input-checkbox.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { LabelledMixin } from '../../mixins/labelled-mixin.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+/**
+ * A checkbox for use in selection components such as lists and tables.
+ * @fires d2l-selection-change - Dispatched when the selected state changes
+ */
+class Checkbox extends SkeletonMixin(LabelledMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * State of the checkbox
+			 */
+			selected: { type: Boolean },
+			/**
+			 * Disables the checkbox
+			 */
+			disabled: { type: Boolean },
+			/**
+			 * Private. Force hovering state of checkbox
+			 */
+			hovering: { type: Boolean },
+			/**
+			 * Key for the selectable
+			 */
+			key: { type: String }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+				line-height: normal;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this._provider = null;
+	}
+
+	get selected() {
+		return this._selected;
+	}
+
+	set selected(val) {
+		const oldVal = this._selected;
+		if (oldVal !== val) {
+			this._selected = val;
+			this.requestUpdate('selected', oldVal);
+
+			// dispatch the event for all selected changes (not just when the user interacts directly with the checkbox)
+			this.dispatchEvent(new CustomEvent('d2l-selection-change', {
+				bubbles: true,
+				composed: true,
+				detail: { key: this.key, selected: this._selected }
+			}));
+
+		}
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		// delay subscription otherwise import/upgrade order can cause selection mixin to miss event
+		requestAnimationFrame(() => {
+			const evt = new CustomEvent('d2l-selection-checkbox-subscribe', {
+				bubbles: true,
+				composed: true,
+				detail: {}
+			});
+			this.dispatchEvent(evt);
+			this._provider = evt.detail.provider;
+		});
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (!this._provider) return;
+		this._provider.unsubscribeSelectable(this);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.key || this.key.length === 0) console.warn('d2l-selection-checkbox component requires a key.');
+		if (!this.label || this.label.length === 0) console.warn('d2l-selection-checkbox component requires label text.');
+	}
+
+	render() {
+		return html`
+			<d2l-input-checkbox
+				aria-label="${this.label}"
+				@change="${this._handleChange}"
+				?checked="${this.selected}"
+				class="${ifDefined(this.hovering ? 'd2l-hovering' : undefined)}"
```

Comment:
If you just applied the `checkboxStyles` to a native checkbox here I think you could constrain the "hovering" hackiness to just lists + selection-checkbox, but I'm not sure what other goodies you'd lose (like skeletons).

Reply:
That's true - I think it is just the skeleton that we'd lose, but it could be handled as well.  I'll fix the other stuff first.  😄  I don't feel super strongly about this one, although I don't want anyone else to use this.